### PR TITLE
fixes rounding issue in automatablemodel

### PR DESCRIPTION
**Pull request by BaraMGB**
fixes https://github.com/LMMS/lmms/pull/3075#discussion_r114059629

### CODE REVIEW

**Comment by PhysSong** [comment]
Tested. It works well.

**Comment by Umcaruje** [comment]
Can you adress @zapashcanon's comment from here too: https://github.com/LMMS/lmms/pull/3075#discussion_r119985704

Tested it out, works like a charm

**Comment by BaraMGB** [comment]
@umcaruje yes, I noticed it. I'll change that line. 👍

File: src/gui/widgets/Knob.cpp [hunk 1]
```diff
@@ -749,7 +748,7 @@ void Knob::setPosition( const QPoint & _p )
 	{
 		if( qAbs( value ) >= step )
 		{
-			float roundedValue = static_cast<float>( static_cast<int>( ( oldValue - value ) / step + 0.5 ) ) * step;
+			float roundedValue = qRound( ( oldValue - value ) / step ) * step;
```
Comment (zapashcanon) [inline src/gui/widgets/Knob.cpp]: All this code:

```c++
		if( qAbs( value ) >= step )
		{
			float roundedValue = qRound( ( oldValue - value ) / step ) * step;
			model()->setValue( roundedValue );
			m_leftOver = 0.0f;
		}
		else
		{
			m_leftOver = value;
		}
```

is duplicated, maybe something like:

```c++
	float consideredValue = value; // linear code

	if( model()->isScaleLogarithmic() ) // logarithmic code
	{
		const float pos = model()->minValue() < 0
			? oldValue / qMax( qAbs( model()->maxValue() ), qAbs( model()->minValue() ) )
			: ( oldValue - model()->minValue() ) / model()->range();
		const float ratio = 0.1f + qAbs( pos ) * 15.f;
		consideredValue = value * ratio;
	}

	if( qAbs( consideredValue ) >= step )
	{
		float roundedValue = qRound( ( oldValue - value ) / step ) * step;
		model()->setValue( roundedValue );
		m_leftOver = 0.0f;
	}
	else
	{
		m_leftOver = value;
	}
```

**Comment by tresf** [comment]
Cherry-picked to master via 902493b.  Please target non-essential code cleanup at the master branch, once this fix is merged in.

**Pull request merged**